docs/vercel/src/Index.svelte

```diff
@@ -1,41 +1,62 @@
 <script lang="ts">
   import AsciiPreview from './lib/AsciiPreview.svelte';
   import data from '../../../languages.yaml';
-  import type { Languages } from '../../../languages.yaml';
+  import type { Languages, Language } from '../../../languages.yaml';
   import { onMount } from 'svelte';
+  import { writable, derived } from 'svelte/store';
 
-  let tag_name: string;
-  let html_url: string;
+  let tagName: string;
+  let htmlUrl: string;
+  let menuOpen: boolean;
 
-  const languages = Object.entries(data as Languages).map(
-    ([name, { ascii, colors }]) => ({
+  const languages: Language[] = Object.entries(data as Languages).map(
+    ([name, { type, ascii, colors }]) => ({
       name,
+      type,
       ascii,
-      ...colors,
+      colors,
     })
   );
 
+  const languageTypes = new Set<string>(
+    Object.values(data as Languages).map(({ type }) => type)
+  );
+
+  const filter = writable({
+    checkboxes: [] as string[],
+  });
+
+  const filteredLanguages = derived(filter, ($filter) => {
+    let filtered: Language[] = languages;
+    if ($filter.checkboxes.length > 0) {
+      filtered = filtered.filter((d: Language) =>
+        $filter.checkboxes.includes(d.type)
+      );
+    }
+    return filtered;
+  });
```

Comment:
I'm more of a Vue person than a Svelte person, but in Vue I would've probably bound the checkbox values to a `{ [key: string]: boolean }` instead of a `string[]`. Then the filtered languages could be `languages.filter(({ type }) => $filter.checkboxes[type])`.

Reply:
If I do this I won't be able to use the `bind:group`  on the input element easily. I'm also not very familiar with Svelte.